File: JavaScript/uri-1045.js
// URI - 1045: https://www.urionlinejudge.com.br/judge/pt/problems/view/1045

const input = require('fs').readFileSync('/dev/stdin', 'utf-8');
const sides = input.split(' ').map(side => parseFloat(side));
const triangleTypes = getTriangleTypeOf(sides.sort((a, b) => b - a));
triangleTypes.forEach(type => console.log(type));

function getTriangleTypeOf(sides) {
  const types = new Array();
  if (isTriangle(sides)) {
    if (isRightTriangle(sides))
      types.push('TRIANGULO RETANGULO');
    if (isObtusangleTriangle(sides))
      types.push('TRIANGULO OBTUSANGULO');
    if (isAcutanguloTriangle(sides))
      types.push('TRIANGULO ACUTANGULO');
    if (isEquilateralTriangle(sides))
      types.push('TRIANGULO EQUILATERO');
    else if (isIsoscelesTriangle(sides))
      types.push('TRIANGULO ISOSCELES');
  }
  else 
    types.push('NAO FORMA TRIANGULO');
  return types.slice(0);
}

function isTriangle(sides) {
  return sides[0] < (sides[1] + sides[2]);
}

function isRightTriangle(sides) {
  return Math.pow(sides[0], 2) === (Math.pow(sides[1], 2) + Math.pow(sides[2], 2));
}

function isObtusangleTriangle(sides) {
  return Math.pow(sides[0], 2) > (Math.pow(sides[1], 2) + Math.pow(sides[2], 2));
}

function isAcutanguloTriangle(sides) {
  return Math.pow(sides[0], 2) < (Math.pow(sides[1], 2) + Math.pow(sides[2], 2));
}

function isEquilateralTriangle(sides) {
  return sides[0] === sides[1] && sides[0] === sides[2] && sides[1] === sides[2];
}

function isIsoscelesTriangle(sides) {
  return sides[0] === sides[1] || sides[0] === sides[2] || sides[1] === sides[2];
}

File: Ruby/uri-1045.rb
# frozen_string_literal: true

# URI - 1045: https://www.urionlinejudge.com.br/judge/pt/problems/view/1045

def type_of_this_triangle(triangle)
  types = []
  not_triangle?(triangle, types)
  if types.empty?
    right_triangle?(triangle, types)
    triangle_obtusangulo?(triangle, types)
    acutangle_triangle?(triangle, types)
    equilateral_triangle?(triangle, types)
    isosceles_triangle?(triangle, types) if types.last != 'TRIANGULO EQUILATERO'
  end
  types
end

def not_triangle?(triangle, types)
  types << 'NAO FORMA TRIANGULO' if triangle[0] >= (triangle[1] + triangle[2])
end

def right_triangle?(triangle, types)
  types << 'TRIANGULO RETANGULO' if triangle[0]**2 == (triangle[1]**2 + triangle[2]**2)
end

def triangle_obtusangulo?(triangle, types)
  types << 'TRIANGULO OBTUSANGULO' if triangle[0]**2 > (triangle[1]**2 + triangle[2]**2)
end

def acutangle_triangle?(triangle, types)
  types << 'TRIANGULO ACUTANGULO' if triangle[0]**2 < (triangle[1]**2 + triangle[2]**2)
end

def equilateral_triangle?(triangle, types)
  types << 'TRIANGULO EQUILATERO' if triangle[0] == triangle[1] && triangle[0] == triangle[2]
end

def isosceles_triangle?(triangle, types)
  types << 'TRIANGULO ISOSCELES' if triangle[0] == triangle[1] ||
                                    triangle[0] == triangle[2] ||
                                    triangle[1] == triangle[2]
end

triangle = gets.split(' ').map(&:to_f)
puts type_of_this_triangle(triangle.sort { |a, b| b <=> a })
